subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -173,6 +192,16 @@ private void writeProperties(File propertiesFileDestination) {
         }
     }
 
+    private String getDistributionSha256Sum(File propertiesFileDestination) {
+        if (distributionSha256Sum != null) {
+            return distributionSha256Sum;
+        } else if (propertiesFileDestination.exists()) {
```

Comment:
❓ I understand it is covered in the `checkProperties` method, but just to be defensive, would it make sense to guard against version mismatches here?
```suggestion
        } else if (GradleVersion.current().equals(gradleVersion)  && propertiesFileDestination.exists()) {
```